src/constructRoutes.js

```diff
@@ -23,22 +25,26 @@ import { inBrowser } from "./environment-helpers.js";
  * mode: string;
  * base: string;
  * containerEl: ContainerEl;
- * routes: Array<RouteChild>;
+ * routes: Array<ResolvedRouteChild>;
  * }} ResolvedRoutesConfig
  *
- * @typedef {UrlRoute | Application | Node | HTMLElementDescription} RouteChild
+ * @typedef {UrlRoute | Application | Node} RouteChild
  *
- * @typedef {{
- * type: string;
- * [attributeName: string]: string;
- * }} HTMLElementDescription
+ * @typedef {ResolvedUrlRoute | Application | Node} ResolvedRouteChild
  *
  * @typedef {string | HTMLElement | import('parse5').Element} ContainerEl
  *
  * @typedef {{
  * type: string;
  * path: string;
  * routes: Array<Route>;
+ * activeWhen: import('single-spa').ActivityFn;
```

Comment:
This is one of the most important conceptual changes here - the resolved routes now have an "activeWhen" function that can be called with a URL/Location at any time to check whether they are active.